src/super_gradients/training/dataloaders/adapters.py

```diff
@@ -0,0 +1,154 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Iterable
+
+import torch
+
+from data_gradients.dataset_adapters.config.data_config import DataConfig
+from data_gradients.dataset_adapters.base_adapter import BaseDatasetAdapter
+from data_gradients.dataset_adapters.config.typing_utils import SupportedDataType
+
+from super_gradients.training.utils.collate_fn import DetectionCollateFN
+from super_gradients.training.utils.collate_fn.adapters import (
+    BaseDatasetAdapterCollateFN,
+    ClassificationDatasetAdapterCollateFN,
+    DetectionDatasetAdapterCollateFN,
+    SegmentationDatasetAdapterCollateFN,
+)
+
+
+class BaseDataloaderAdapterFactory(ABC):
+    """Factory class, responsible for adapting datasets/dataloaders to seamlessly work with SG format."""
+
+    @classmethod
+    def from_dataset(
+        cls,
+        dataset: torch.utils.data.Dataset,
+        adapter_config: Optional[DataConfig] = None,
+        adapter_cache_path: Optional[str] = None,
+        collate_fn: Optional[callable] = None,
+        **dataloader_kwargs,
+    ) -> torch.utils.data.DataLoader:
+        """Wrap a DataLoader to adapt its output to fit SuperGradients format for the specific task.
+
+        :param dataset:                 Dataset to adapt.
+        :param adapter_config:          Adapter configuration. Use this if you want to hard code some specificities about your dataset.
```

Comment:
`hardcode` => explicitly set
`specificities` => specific params
about your dataset => of your dataset

Reply:
[975528b](https://github.com/Deci-AI/super-gradients/pull/1471/commits/975528bdc716f36cf4f3c472c887f95b6e7d2fd8)